src/Components/Server/src/Circuits/RemoteRenderer.cs

```diff
@@ -116,79 +135,166 @@ protected override Task UpdateDisplayAsync(in RenderBatch batch)
             //       buffer on every render.
             var batchBytes = MessagePackSerializer.Serialize(batch, RenderBatchFormatterResolver.Instance);
 
-            if (!_client.Connected)
-            {
-                // Buffer the rendered batches while the client is disconnected. We'll send it down once the client reconnects.
-                OfflineRenderBatches.Enqueue(batchBytes);
-                return Task.CompletedTask;
-            }
+            var renderId = Interlocked.Increment(ref _nextRenderId);
+
+            var pendingRender = new PendingRender(
+                renderId,
+                batchBytes,
+                new TaskCompletionSource<object>());
+
+            // Buffer the rendered batches no matter what. We'll send it down immediately when the client
+            // is connected or right after the client reconnects.
+
+            PendingRenderBatches.Enqueue(pendingRender);
+
+            // Fire and forget the initial send for this batch (if connected). Otherwise it will be sent
+            // as soon as the client reconnects.
+            var _ = WriteBatchBytesAsync(pendingRender);
```

Comment:
@SteveSandersonMS I'm not going to do it right now, but I would hope that we add something in the future to track these tasks if desired. Firing an event with the task should be cheap enough.